packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit_webview_widget.dart

```diff
@@ -451,11 +483,67 @@ class WebKitWebViewPlatformController extends WebViewPlatformController {
         },
       );
     } else {
-      webView.observeValue = null;
+      webView.setObserveValue(null);
       return webView.removeObserver(webView, keyPath: 'estimatedProgress');
     }
   }
 
+  void _setJavaScriptMode(JavascriptMode mode) {
+    switch (mode) {
+      case JavascriptMode.disabled:
+        webView.configuration.preferences.setJavaScriptEnabled(false);
+        break;
+      case JavascriptMode.unrestricted:
+        webView.configuration.preferences.setJavaScriptEnabled(true);
+        break;
+    }
+  }
+
+  void _setUserAgent(WebSetting<String?> userAgent) {
+    if (userAgent.isPresent) {
+      webView.setCustomUserAgent(userAgent.value);
+    }
+  }
+
+  Future<void> _setZoomEnabled(bool zoomEnabled) async {
+    if (_zoomEnabled == zoomEnabled) {
+      return;
+    }
+
+    _zoomEnabled = zoomEnabled;
+    if (!zoomEnabled) {
+      await _disableZoom();
+    } else {
+      // WkWebView does not support removing a single user script, so instead we
+      // remove all user scripts, all message handlers, and re-register channels
+      // that shouldn't be removed. Note that this workaround could interfere
+      // with exposing support for custom scripts from applications.
+      webView.configuration.userContentController.removeAllUserScripts();
+      webView.configuration.userContentController
+          .removeAllScriptMessageHandlers();
+
+      final Set<String> javaScriptChannelNames =
+          _scriptMessageHandlers.keys.toSet();
+      _scriptMessageHandlers.clear();
+
+      await addJavascriptChannels(javaScriptChannelNames);
+    }
+  }
+
+  Future<void> _disableZoom() {
+    const WKUserScript userScript = WKUserScript(
+      "var meta = document.createElement('meta');"
+      "meta.name = 'viewport';"
+      "meta.content = 'width=device-width, initial-scale=1.0, maximum-scale=1.0,"
+      "user-scalable=no';"
+      "var head = document.getElementsByTagName('head')[0];head.appendChild(meta);",
+      WKUserScriptInjectionTime.atDocumentEnd,
+      isMainFrameOnly: true,
+    );
+    return webView.configuration.userContentController
+        .addUserScript(userScript);
+  }
```

Comment:
This is different than the current implementation which calls this script manually every time a page loads: https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FLTWKNavigationDelegate.m#L65.

I think setting a user script once is a better way to do this.

I also tested this method by adding it to the current implementation and it worked.

```objc
NSString *myString = @"var meta = document.createElement('meta');meta.name = 'viewport';meta.content = 'width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no';var head = document.getElementsByTagName('head')[0];head.appendChild(meta);";

WKUserScript *myScript = [[WKUserScript alloc] initWithSource:myString
                            injectionTime:WKUserScriptInjectionTimeAtDocumentEnd
                                                 forMainFrameOnly:YES];
[userContentController addUserScript:myScript];
```

Reply:
Yes, that is better. I completely forgot about user scripts when reviewing the zoom PR.